client.go

```diff
@@ -1989,19 +1990,35 @@ func (c *Client) GetMigrationSourceWS(container string) (*Response, error) {
 	return c.post(url, body, Async)
 }
 
-func (c *Client) MigrateFrom(name string, operation string, certificate string, secrets map[string]string, architecture string, config map[string]string, devices shared.Devices, profiles []string, baseImage string, ephemeral bool) (*Response, error) {
+func (c *Client) MigrateFrom(name string, operation string, certificate string,
```

Comment:
Maybe not necessary as part of this series, but this strikes me that it would benefit from a MigrateFromOpts struct, so we don't have to pass all the stuff explicitly when it's in push mode.